src/index.ts

```diff
@@ -0,0 +1,180 @@
+import express from 'express';
+import { PrismaClient } from '@prisma/client';
+
+const prisma = new PrismaClient();
+const app = express();
+
+app.use(express.json());
+
+// Tạo một user mới
+app.post('/users', async (req, res) => {
+    try {
+        const { name, age } = req.body;
+
+        const user = await prisma.user.create({
+            data: {
+                name: name,
+                age: age,
+            },
+        });
+
+        res.status(201).json(user);
+    } catch (error) {
+        res.status(500).json({ error: 'Something went wrong.' });
+    }
+});
+
+// Lay danh sach user cung voi post tuong ung cho tung user
+app.get('/users', async (req, res) => {
+    try {
+        // Lay ra thong tin tren query param
+        const { page = 1, limit = 2 } = req.query;
+
+        // Parse `page` va `limit` tren query param tha`nh number
+        const pageNumber = parseInt(page as string, 10);
+        const pageSize = parseInt(limit as string, 10);
+
+        const offset = (pageNumber - 1) * pageSize;
+        const users = await prisma.user.findMany(
+            {
+                skip: offset,       // Bo qua cac ban ghi truoc do
+                take: pageSize,     // Lay so luong ban ghi theo gioi han
+                include: {          // Lay danh sach post cho tung user tuong ung
+                    posts: true,
+                },
+            }
+        );
+        const totalUsers = await prisma.user.count();
+
+        res.json({
+            totalUsers,
+            totalPages: Math.ceil(totalUsers / pageSize),
+            currentPage: pageNumber,
+            users,
+        });
+    } catch (error) {
+        res.status(500).json({ error: 'Something went wrong.' });
+    }
+});
+
+// Lay thong tin cua 1 user cu the kem theo tong so luong Post cua user do
+app.get('/users/:id', async (req, res) => {
+    const { id } = req.params;
+    try {
+        const users= await prisma.user.findUnique({
+            where: {id: id},
```

Comment:
[nit] e comment những chỗ như này đến lần thứ 2 rồi. anh chú ý tránh gặp lại

```suggestion
            where: {id},
```

Reply:
Done